pkg/ctrlmgr/controllers/cluster/cluster_controller.go

```diff
@@ -138,18 +164,50 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	// start to scout loop for memberCluster warden, non-block
 	err = multicluster.Interface().ScoutFor(context.Background(), currentCluster.Name)
 	if err != nil {
-		log.Error("start scout for memberCluster %v failed", currentCluster.Name)
+		log.Error("start scout for cluster %v failed", currentCluster.Name)
+		return ctrl.Result{}, err
 	}
 
 	// update cluster status to processing
-	err = r.updateClusterStatus(ctx, currentCluster)
+	updateFn := func(cluster *clusterv1.Cluster) {
+		state := clusterv1.ClusterProcessing
+		reason := fmt.Sprintf("cluster %s initializing", cluster.Name)
+		cluster.Status.State = &state
+		cluster.Status.Reason = reason
+	}
+
+	err = utils.UpdateStatus(ctx, r.Client, &currentCluster, updateFn)
 	if err != nil {
-		log.Error("update cluster %v status failed", currentCluster.Name)
+		log.Warn("update cluster %v status failed: %v", currentCluster.Name, err)
 	}
 
 	return ctrl.Result{}, nil
 }
 
+// It enqueues a cluster for later reconciliation. This occurs in a goroutine
+// so the caller doesn't block; since the reconciler is never garbage-collected, this is safe.
+func (r *ClusterReconciler) enqueue(cluster clusterv1.Cluster) {
+	const retrySeconds = 7
+
+	config, _ := kubeconfig.LoadKubeConfigFromBytes(cluster.Spec.KubeConfig)
```

Comment:
need err and when fix cluster.spec.kubeconfig right it will not reconnect

Reply:
i think it‘s neccessary to set _LoadKubeConfig_ to _time.tick_ before _client.New_，in case cluster was changed and _time.tick_ not know.

Reply:
1. kubeconfig.LoadKubeConfigFromBytes() was be called before, there must be no error when going here.
2. cluster cr don't allow update operate

Reply:
we need consider that we should release the goroutine when cluster cr was been deleted